signer/src/bitcoin/rpc.rs

```diff
@@ -60,77 +56,161 @@ pub struct GetTxResponse {
+    pub fee: Option<Amount>,
     /// The raw bitcoin transaction.
     #[serde(with = "bitcoin::consensus::serde::With::<bitcoin::consensus::serde::Hex>")]
     #[serde(rename = "hex")]
     pub tx: Transaction,
-    /// The transaction id (the same value provided in the RPC).
-    pub txid: Txid,
-    /// The transaction hash (differs from txid for witness transactions).
-    pub hash: Wtxid,
-    /// The serialized transaction size.
-    pub size: u64,
-    /// The virtual transaction size (differs from size for witness
-    /// transactions).
-    pub vsize: u64,
     /// The inputs into the transaction.
     pub vin: Vec<BitcoinTxVin>,
-    /// A description of the transactions outputs. This object is missing
-    /// the `desc` field in the `scriptPubKey` object. That field is the
-    /// "Inferred descriptor for the output".
-    pub vout: Vec<BitcoinTxInfoVout>,
-    /// The block hash of the Bitcoin block that includes this transaction.
-    #[serde(rename = "blockhash")]
-    pub block_hash: BlockHash,
-    /// The number of confirmations deep from that chain tip of the bitcoin
-    /// block that includes this transaction.
-    pub confirmations: u32,
-    /// The Unix epoch time when the block was mined. It reflects the
-    /// timestamp as recorded by the miner of the block.
-    #[serde(rename = "blocktime")]
-    pub block_time: u64,
+}
+
+/// A struct with more detailed information for transaction inputs. It is
+/// returned in responses to bitcoin core's `getrawtransaction` and
+/// `getblock` RPCs.
+///
+/// This struct omits several fields: `scriptSig`, `txinwitness`, for
+/// non-coinbase transactions, the `coinbase` field for coinbase
+/// transactions, and the `sequence` field. See
+/// [`bitcoincore_rpc_json::GetRawTransactionResultVin`] or
+/// https://bitcoincore.org/en/doc/25.0.0/rpc/rawtransactions/getrawtransaction/
+/// for more information on what is missing.
+#[derive(Clone, PartialEq, Eq, Debug, serde::Deserialize, serde::Serialize)]
+pub struct BitcoinTxVin {
+    /// The transaction ID. Not provided for coinbase transactions.
+    pub txid: Option<Txid>,
+    /// The output index. Not provided for coinbase transactions.
+    pub vout: Option<u32>,
+    /// The previous output.
+    ///
+    /// For non-coinbase transactions, this field is omitted if block undo
+    /// data is not available, so it is missing whenever the `fee` field is
+    /// missing in the [`BitcoinTxInfo`]. It is always omitted for coinbase
+    /// transactions.
+    pub prevout: Option<BitcoinTxVinPrevout>,
+}
+
+/// The previous output, omitted if block undo data is not available.
+///
+/// The `height` field is included in the actual response, but omitted
+/// here.
+#[derive(Clone, PartialEq, Eq, Debug, serde::Deserialize, serde::Serialize)]
+pub struct BitcoinTxVinPrevout {
+    /// Whether the previous output is from a coinbase transaction or not.
+    pub generated: bool,
```

Comment:
Maybe `is_coinbase` will be better naming here ?

Reply:
I started to rename it, but then realized that it is unnecessary, so I removed the field instead.

Reply:
https://github.com/stacks-sbtc/sbtc/pull/1642/commits/f1fbc5ba6628b80c04b72bee784ad895e6a3573a